simapp/app.go

```diff
@@ -35,22 +38,22 @@ var (
 	// default home directories for the application daemon
 	DefaultNodeHome = os.ExpandEnv("$HOME/.simapp")
 
-	// The ModuleBasicManager is in charge of setting up basic,
+	// The module BasicManager is in charge of setting up basic,
 	// non-dependant module elements, such as codec registration
 	// and genesis verification.
-	ModuleBasics sdk.ModuleBasicManager
+	ModuleBasics module.BasicManager
 )
 
 func init() {
-	ModuleBasics = sdk.NewModuleBasicManager(
+	ModuleBasics = module.NewBasicManager(
 		genaccounts.AppModuleBasic{},
 		genutil.AppModuleBasic{},
 		auth.AppModuleBasic{},
 		bank.AppModuleBasic{},
 		staking.AppModuleBasic{},
 		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
-		gov.AppModuleBasic{},
+		gov.NewAppModuleBasic(paramsclient.ProposalHandler, distrclient.ProposalHandler),
```

Comment:
Again, using `init` is an antipattern. Also do we need constructors for all of the other `AppModuleBasic` implementations?

Reply:
no we only need constructors for gov.